com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationProductizationStateUpdater.java

```diff
@@ -0,0 +1,37 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaApplication;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+
+public abstract class ApplicationProductizationStateUpdater {
+
+    private List<DeployedMtaApplication> deployedModules;
```

Comment:
This state is only used in the `updateIdleDeployedApplications` method. Why not pass it as an argument to the method itself?